IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/WebElement.cs

```diff
@@ -1,88 +1,118 @@
 ﻿using OpenQA.Selenium;
-using OpenQA.Selenium.Support.UI;
-using System;
-using System.Collections.Generic;
-using System.Collections.ObjectModel;
-using System.Drawing;
 using System.Linq;
-using System.Text;
+using System.Drawing;
 using System.Threading.Tasks;
+using System.Collections.ObjectModel;
 
 namespace IntelliTect.TestTools.Selenate
 {
     public class WebElement : IWebElement
     {
-        public WebElement(IWebElement wrappedElement, By by, IWebDriver driver)
+        // Move this class to Extensions at some point. This is really no longer needed with how easy it is to call waits now. This would just be for people that want an "out of the box" implementation
+        // Also: do we ever need to re-find the element if an operation throws a stale element reference?
+        //We were doing that in the prior impelmentation, but I'm not sure actually sure it's needed since we're making sure the element is found before doing anything with it.
+        public WebElement(IWebElement wrappedElement, /*By by,*/ IWebDriver driver)
         {
             WrappedElement = wrappedElement;
-            By = by;
+            //By = by;
             _Driver = driver;
         }
 
         public By By { get; }
-        public bool Initialized => WrappedElement != null;
 
-        // Probably wap out the Task<> properties in favor of Get methods for consistency.
-        // Or scrap this class entirely once retry logic is abstracted out into a single callable method
         string IWebElement.TagName => TagName.ConfigureAwait(false).GetAwaiter().GetResult();
-        public Task<string> TagName => RetryAction("tagname");
+        public Task<string> TagName
```

Comment:
This should really be a method not a property. Properties are expected to be fast quick access to data, not long running async calls.

Same comment from the other properties.

Reply:
Okay, I was kinda wondering about that. It seemed to be a bit odd/misleading for the property to simply call into a method that's doing a bunch of stuff. I'm considering re-doing this entire class now as it's starting to depart from the default WebElement implementation. I'm not sure what to do about that, as I like implementing IWebElement to maintain some level of compatibility with Selenium, but at the same time the class is really getting beyond what Selenium does.

I'll put this in the task list and think about it, maybe discuss it a bit with some people.